packages/labs/src/components/omnibox/omnibox.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import {
+    AbstractComponent,
+    Classes as CoreClasses,
+    HTMLInputProps,
+    IBackdropProps,
+    IInputGroupProps,
+    InputGroup,
+    IOverlayableProps,
+    IOverlayProps,
+    Overlay,
+    Utils,
+} from "@blueprintjs/core";
+import {
+    IListItemsProps,
+    IQueryListRendererProps,
+    ISelectItemRendererProps,
+    QueryList,
+} from "../";
+import * as Classes from "../../common/classes";
+
+export interface IOmniboxProps<T> extends IListItemsProps<T> {
+    /**
+     * Custom renderer for an item in the dropdown list. Receives a boolean indicating whether
+     * this item is active (selected by keyboard arrows) and an `onClick` event handler that
+     * should be attached to the returned element.
+     */
+    itemRenderer: (itemProps: ISelectItemRendererProps<T>) => JSX.Element;
+
+    /** React child to render when filtering items returns zero results. */
+    noResults?: string | JSX.Element;
```

Comment:
If you don't mind, I can do this in a follow up PR across `Omnibox`, `Select` and `MultiSelect`